Sources/Common/DataModel/Cylinder/index.js

```diff
@@ -0,0 +1,103 @@
+import macro from 'vtk.js/Sources/macro';
+import vtkMath from 'vtk.js/Sources/Common/Core/Math';
+
+// ----------------------------------------------------------------------------
+// Global methods
+// ----------------------------------------------------------------------------
+
+function evaluate(radius, center, axis, x) {
+  const x2C = new Float32Array(3);
+  x2C[0] = x[0] - center[0];
+  x2C[1] = x[1] - center[1];
+  x2C[2] = x[2] - center[2];
+
+  const proj = vtkMath.dot(axis, x2C);
+
+  const retVal = ((vtkMath.dot(x2C, x2C) - (proj * proj)) - (radius * radius));
+
+  return retVal;
+}
+
+// ----------------------------------------------------------------------------
+// Static API
+// ----------------------------------------------------------------------------
+
+export const STATIC = {
+  evaluate,
+};
+
+
+// ----------------------------------------------------------------------------
+// vtkCylinder methods
+// ----------------------------------------------------------------------------
+
+function vtkCylinder(publicAPI, model) {
+  // Set our className
+  model.classHierarchy.push('vtkCylinder');
+
+  publicAPI.evaluateFunction = (xyz) => {
+    const x2C = new Float32Array(3);
```

Comment:
Do you specifically need to be a 32 bit computation?

another way to write that could have been:

```
const x2C = [
   xyz[0] - model.center[0],
   xyz[1] - model.center[1],
   xyz[2] - model.center[2],
];
```